# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1505145588 14400
#      Mon Sep 11 11:59:48 2017 -0400
# Node ID c2ef0eddf6bc29dfaf6bdf189bd88812a0629d37
# Parent  ef85638c605a056cc6c265fc1081b24feec10e25
* symtab.cc (F__current_scope__): Return scope name.

diff --git a/libinterp/corefcn/symtab.cc b/libinterp/corefcn/symtab.cc
--- a/libinterp/corefcn/symtab.cc
+++ b/libinterp/corefcn/symtab.cc
@@ -1965,17 +1965,21 @@ DEFMETHOD (__current_scope__, interp, , 
            doc: /* -*- texinfo -*-
 @deftypefn {} {[@var{scope}, @var{context}]} __current_scope__ ()
 Return the current scope and context as integers.
 @seealso{__dump_symtab_info__}
 @end deftypefn */)
 {
   octave::symbol_table& symtab = interp.get_symbol_table ();
 
-  return ovl (symtab.current_scope (), symtab.current_context ());
+  octave::symbol_table::scope *scope = symtab.current_scope ();
+
+  std::string nm = scope ? scope->name () : "<unknown>";
+
+  return ovl (nm, symtab.current_context ());
 }
 
 DEFMETHOD (__dump_symtab_info__, interp, args, ,
            doc: /* -*- texinfo -*-
 @deftypefn  {} {} __dump_symtab_info__ ()
 @deftypefnx {} {} __dump_symtab_info__ (@var{function})
 Undocumented internal function.
 @seealso{__current_scope__}
